src/components/infinite-scroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * From https://github.com/jaredpalmer/react-simple-infinite-scroll
+ * Updated to accept a scroll container React ref as the "window"
+ */
+import * as React from 'react';
+import throttleFn from 'lodash/throttle';
+
+export type Props = {
+    /** Components to render, should include entities/loading states/end message */
+    children: React.ReactNode;
+    /** Does the resource have more entities */
+    hasMore: boolean;
+    /** Are more entities being loaded */
+    isLoading: boolean;
+    /** Callback to load more entities */
+    onLoadMore: () => void;
+    /** React ref of the scroll container.
+     * Used to listen to scroll events and calculate onLoadMore calls.
+     * Set useWindow to true to use the window as scroll container.
+     */
+    scrollContainerNode?: HTMLElement;
+    /** Scroll threshold */
+    threshold?: number;
+    /** Throttle rate */
+    throttle?: number;
+    /** Set useWindow to true to use the window as scroll container. If set to true, will ignore scrollContainerRef. */
+    useWindow?: boolean;
+};
+
+function InfiniteScroll({
+    children,
+    hasMore,
+    isLoading,
+    onLoadMore,
+    scrollContainerNode,
+    threshold = 100,
+    throttle = 64,
+    useWindow = false,
+}: Props) {
+    const sentinelRef = React.createRef<HTMLDivElement>();
+
+    function getContainer(): EventTarget | undefined {
+        return useWindow ? window : scrollContainerNode;
+    }
+
+    React.useEffect(() => {
+        function onContainerScroll() {
```

Comment:
Would it better to declare this outside of the `React.useEffect`?

Reply:
I don't really think it would beneficial but I don't mind moving outside the component

Reply:
I think persuasion and consensus are really import. Let me try again, and if I'm not persuasive, no worries :) `onContainerScroll` has closure from `InfiniteScroll`. This increases the chances of accidentally making use of sideEffects from `InfiniteScroll`. Moving it out of the closure and requiring is dependencies to be passed as parameters protects against this hazard. It also has the added benefits of making that function testable and reusable. What are the benefits of leaving as is?